controllers/complianceeventsapi/types.go

```diff
@@ -50,10 +70,34 @@ func (ce ComplianceEvent) Validate() error {
 	return errors.Join(errs...)
 }
 
+func (ce *ComplianceEvent) Create(ctx context.Context, db *sql.DB) error {
+	if ce.Event.ClusterID == 0 {
+		ce.Event.ClusterID = ce.Cluster.KeyID
+	}
+
+	if ce.Event.PolicyID == 0 {
+		ce.Event.PolicyID = ce.Policy.KeyID
+	}
+
+	insertQuery, insertArgs := ce.Event.InsertQuery()
+
+	row := db.QueryRowContext(ctx, insertQuery+"RETURNING id", insertArgs...) //nolint:execinquery
```

Comment:
The silly linter should recognize RETURNING...

Another tangent/discussion: it's not as easy as I would like to see the exact query this uses, partly because the `InsertQuery` calls through two other functions... It's nice that multiple things can use `generateInsertQuery`, but I'm wondering about the overall cost/benefit.